### PR TITLE
docs: Updated storybook templates

### LINKED ISSUE #2519: [docs] Storybook updates to match updated USWDS recs
**Does this documentation relate to a specific USWDS feature?**
[Address template](https://designsystem.digital.gov/templates/form-templates/address-form/)
[Name template](https://designsystem.digital.gov/templates/form-templates/name-form/)

**Is your documentation related to a problem or inconsistency? Please describe.**
No

**Describe the documentation you'd like**
Check the USWDS site links above for how they frame the components. Mimic that in storybook.

**Additional context**
This would be very useful for the current project @kleinschmidtj and @werdnanoslen are working on

### PR DESCRIPTION
**Pull request by werdnanoslen**
# Summary

Updates [address](http://localhost:9009/?path=/story/components-form-templates--address-form) and [name](http://localhost:9009/?path=/story/components-form-templates--name-form) storybook templates

## Related Issues or PRs

Resolves #2519 

## How To Test

Open [address](http://localhost:9009/?path=/story/components-form-templates--address-form) and [name](http://localhost:9009/?path=/story/components-form-templates--name-form) and compare to [USWDS templates](https://designsystem.digital.gov/templates/form-templates/)

### Screenshots (optional)
image
image

### CODE REVIEW

**Comment by werdnanoslen** [comment]
No test needed, only story changed

**Comment by werdnanoslen** [comment]
The address template will need to be updated once #2524 is merged, so please merge that one first.

File: src/components/forms/Form/Form.stories.tsx [hunk 1]
```diff
@@ -79,111 +79,136 @@ export const textInputForm = (): React.ReactElement => (
...
     <Fieldset legend="Mailing address" legendStyle="large">
-      <Label htmlFor="mailing-address-1">Street address 1</Label>
+      <p>
+        Required fields are marked with an asterisk (
+        <abbr title="required" className="usa-hint usa-hint--required">
+          *
+        </abbr>
+        ).
+      </p>
+      <Label htmlFor="mailing-address-1">Street address</Label>
```
Comment (gidjin) [inline src/components/forms/Form/Form.stories.tsx:120]: Question: Should `Street Address` be marked as required?

**Review by gidjin — APPROVED** [review]

File: src/components/forms/Form/Form.stories.tsx (hunk 1 above)
Comment (werdnanoslen) [reply src/components/forms/Form/Form.stories.tsx:120]: not necessarily, i was just copying the [USWDS example](https://designsystem.digital.gov/templates/form-templates/address-form/)

**Pull request merged**